src/hooks/useButtonStyles/useButtonStyles.ts

```diff
@@ -0,0 +1,29 @@
+import type { BackgroundProps, ButtonProps } from '@chakra-ui/react'
+import { useColorMode, useTheme } from '@chakra-ui/react'
+import type { SystemStyleFunction } from '@chakra-ui/theme-tools'
+import { theme } from 'theme/theme'
+
+type UseButtonStyleProps = {
+  variant: string
+  colorScheme: string
+}
+
+export const useButtonStyles = ({ variant, colorScheme }: UseButtonStyleProps) => {
+  const {
+    components: {
+      Button: { variants },
+    },
+  } = useTheme()
+
+  const { [variant]: buttonVariant } = variants as Record<string, SystemStyleFunction>
+
+  const { colorMode } = useColorMode()
+
+  const styleProps = buttonVariant({
+    colorScheme,
+    theme,
+    colorMode,
```

Comment:
The guts of this convenience hook, currying the `buttonVariant` call with these two